custom-recipes/nlp-analysis-ontology-tagging/recipe.py

```diff
@@ -15,21 +14,20 @@
     lang, lang_column = settings.language, ""
 
 tagger = Tagger(
-    settings.text_input,
-    settings.ontology_input,
+    settings.text_input.get_dataframe(),
+    settings.ontology_input.get_dataframe(),
```

Comment:
I would suggest executing these outside of the Tagger context. This way, if an error happens, the stack trace will show more clearly that it's not related to the Tagger.

Also, would be good to load only the columns which are needed for the ontology input

Reply:
here : 9deaf69

Reply:
Almost there, but to load only the columns needed for the ontology input, you can use the `columns` argument in `get_dataframe`: https://doc.dataiku.com/dss/latest/python-api/datasets-reference.html?highlight=get_dataframe#dataiku.Dataset.get_dataframe
It avoids streaming the entire dataset in case you need only a few columns. Here the ontology dataset is small, so the impact is negligible. But in general, it's a good reflex to adopt.

Also, would be good to deactivate `infer_with_pandas` in `get_dataframe` for both datasets. This inference is known to cause issues downstream if the users wants to join on specific columns and the types have been changed.